tests/unit/assessment/__init__.py

```diff
@@ -24,8 +25,9 @@ def _cluster_policy(policy_id: str):
     return Policy(description=definition, policy_family_definition_overrides=overrides)
 
 
-def workspace_client_mock(clusters="no-spark-conf.json"):
+def workspace_client_mock(clusters="no-spark-conf.json", pipeline_cluster="pipeline_cluster.json"):
     ws = create_autospec(WorkspaceClient)
     ws.clusters.list.return_value = _load_list(ClusterDetails, f"../assessment/clusters/{clusters}")
     ws.cluster_policies.get = _cluster_policy
+    ws.pipelines.get().spec.clusters = _load_list(PipelineCluster, f"clusters/{pipeline_cluster}")
```

Comment:
this is silly and makes no sense at all. mock the entire response, not just a subset - `ws.pipelines.get.return_value = _load_list(...`

Reply:
Now capturing the entire PipelineResponse instead of only mocking the cluster part.